vyper/venom/passes/branch_optimization.py

```diff
@@ -27,6 +37,11 @@ def _optimize_branches(self) -> None:
             if cost_a >= cost_b and prev_inst.opcode == "iszero":
                 new_cond = prev_inst.operands[0]
                 term_inst.operands = [new_cond, term_inst.operands[2], term_inst.operands[1]]
+            elif cost_a >= cost_b and prefer_iszero(prev_inst):
+                new_cond = fn.get_next_variable()
+                inst = IRInstruction("iszero", [term_inst.operands[0]], output=new_cond)
+                bb.insert_instruction(inst, index=-1)
+                term_inst.operands = [new_cond, term_inst.operands[2], term_inst.operands[1]]
```

Comment:
the code for this branch is duplicated with the next branch, so we can just fix the condition.
```suggestion
            elif cost_a > cost_b or (cost_a >= cost_b and prefer_iszero(prev_inst)):
                new_cond = fn.get_next_variable()
                inst = IRInstruction("iszero", [term_inst.operands[0]], output=new_cond)
                bb.insert_instruction(inst, index=-1)
                term_inst.operands = [new_cond, term_inst.operands[2], term_inst.operands[1]]
```